docs.feldera.com/docs/pipelines/latency.md

```diff
@@ -0,0 +1,211 @@
+and understand the end-to-end processing latency of your pipelines.
+
+## Background: Understanding pipeline latency
+
+A Feldera pipeline processes data in several stages:
+
+* **Input connectors** read data from the data source (e.g., a database table or a Kafka stream) and parse it
+  into a stream of data change events (inserts, deletes, and updates to SQL tables).
+* **Input queues** buffer incoming data until the SQL engine is ready to
+  process it. If the pipeline keeps up with data arrival, input queues stay nearly empty. When data arrives
+  faster than the engine can process it, the queue fills up, leading to higher latency.
+* **SQL engine** repeatedly dequeues batches of changes from
+  input queues and computes updates to all affected output views.
+* **Output queues** buffer outputs produced by the engine until output connectors process them. If an output
+  connector keeps up with changes, its queue is nearly empty. Otherwise, changes accumulate in the queue,
+  leading to higher end-to-end latencies.
+* **[Output buffers](/connectors#configuring-the-output-buffer)** can be optionally configured for output
+  connectors to batch multiple updates before sending them to the connector. They increase the end-to-end
+  latency by delaying outputs up to a user-configurable bound.
+* **Output connectors** serialize output updates and send them to target data sinks.
+
+![Understanding pipeline latency](latency.png)
+
+## Latency metrics
+
+A pipeline exposes key latency metrics through its [`/metrics`](https://docs.feldera.com/api/retrieve-circuit-metrics-of-a-running-or-paused-pipeline/)
+endpoint.  See complete list of metrics [here](/operations/metrics).
+
+### Completion latency (`input_connector_completion_latency_seconds`)
+
+**Scope**: Per input connector
+
+**Definition**: End-to-end latency, from ingestion of a batch by the input connector until all resulting outputs have been written to all sinks.
+
+**Reported as**: Histogram over the last 600 seconds or 10,000 batches (whichever is smaller).
+
+Note that the granularity of a batch is determined by the input connector.
+
+**Example:**
+
+```
+input_connector_completion_latency_seconds_bucket{pipeline="pipeline-019954ea-adae-70d1-bd40-981514fb2214",endpoint="test_connector.unnamed-0",le="0"} 0
+<skipped>
+input_connector_completion_latency_seconds_bucket{pipeline="pipeline-019954ea-adae-70d1-bd40-981514fb2214",endpoint="test_connector.unnamed-0",le="0.199999"} 13
+input_connector_completion_latency_seconds_bucket{pipeline="pipeline-019954ea-adae-70d1-bd40-981514fb2214",endpoint="test_connector.unnamed-0",le="0.299999"} 40
+input_connector_completion_latency_seconds_bucket{pipeline="pipeline-019954ea-adae-70d1-bd40-981514fb2214",endpoint="test_connector.unnamed-0",le="0.399999"} 54
+<skipped>
+input_connector_completion_latency_seconds_bucket{pipeline="pipeline-019954ea-adae-70d1-bd40-981514fb2214",endpoint="test_connector.unnamed-0",le="inf"} 70
```

Comment:
can an explanation be given how to interpret this?
what is "le"? what is "inf"? what is "70"

Reply:
maybe a graphical representation of the histogram would help?